course/views.py

```diff
@@ -58,10 +60,9 @@ def get_common_objects(self):
                     my_instances.append(instance)
             
             for instance in user.userprofile.enrolled.all().filter(ending_time__gte=end_threshold):
-                if instance not in my_instances:
+                if instance not in my_instances and instance in all_instances:
```

Comment:
Here is a check `instance in <queryset>`. Does it generate a sql query per evaluation?

As an alternative solution, should `user.userprofile.enrolled.all()` limit the output like the `get_visible(user)` does?

Reply:
Seems to me, that only one query is created from that line, even with several enrolled courses.

Reply:
Ok good. Then I guess the django sql cache works as it's supposed to.